### PR TITLE
SDK v1

### PR DESCRIPTION
**Pull request by antho1404**
#### Write
- useAcceptAuction
- useAcceptOffer
- useAddFund (to remove)
- useApproveCollection (internal)
- useApproveCurrency (internal)
- useCancelOffer
- useCreateAuction
- useCreateNFT
- useCreateOffer
- useIPFSUploader
- useInvitation
- useUpdateAccount
- useVerifyAccount

#### Read
- useAuctionStatus
- useBalance (to remove use wagmi)
- useCheckOwnership (internal)
- useConfig (to confirm if still needed, if needed, should be internal to the core package)

#### Auth
- useAuthenticate
- useIsLoggedIn

#### Core package

- [x] Create base core package https://github.com/liteflow-labs/liteflow-js/pull/142
- [x] Create exchange module
  - [x] List token https://github.com/liteflow-labs/liteflow-js/pull/142
  - [x] buyToken https://github.com/liteflow-labs/liteflow-js/pull/144
  - [x] Cancel listing https://github.com/liteflow-labs/liteflow-js/pull/144
  - [x] Place bid https://github.com/liteflow-labs/liteflow-js/pull/142
  - [x] Accept bid https://github.com/liteflow-labs/liteflow-js/pull/144
  - [x] Cancel bid https://github.com/liteflow-labs/liteflow-js/pull/144
  - [x] Create auction https://github.com/liteflow-labs/liteflow-js/pull/159
  - [x] Accept/Finalize auction https://github.com/liteflow-labs/liteflow-js/pull/159
- [x] Create asset module
  - [x] create asset (embed the IPFS uploader) https://github.com/liteflow-labs/liteflow-js/pull/160
- [x] Create account module
  - [ ] Create invitation (to discuss if needed)
  - [x] Update account https://github.com/liteflow-labs/liteflow-js/pull/161
  - [x] Ask for verification https://github.com/liteflow-labs/liteflow-js/pull/161

#### React package 

The react package should have the same interface, but use the `@liteflow/core` package and manage the session with its Context (`useAuthenticate` + `useIsLoggedIn`)

- [x] Remove email connector package https://github.com/liteflow-labs/liteflow-js/pull/162
- [x] Rename `@nft/hooks` to `@liteflow/react` https://github.com/liteflow-labs/liteflow-js/pull/162
- [x] Remove utils and unused hooks https://github.com/liteflow-labs/liteflow-js/pull/162

## How to test

(if needed `npm i ./packages/core -w packages/hooks`)

- `npm run build -w packages/core`
- `npm run build -w packages/react`

or 

- `npm run watch`

In the starter kit run `npm i ./path-to-liteflow-js/packages/react`

#### Testing:

To test correctly the changes, we need to update the starter kit with the new library (this PR contains the fixes for all the coming breaking changes https://github.com/liteflow-labs/starter-kit/pull/319),  we need to test the following
| Scenario | Pass (yes/no) |
| --- | --- |
| **Listing** | |
| Can create a listing | OK |
| Can list for a specific price | OK |
| Can list with a specific expiration date | OK |
| Can list with a native token | OK |
| Can list with an ERC20 | OK |
| Can purchase with native token | OK |
| Can purchase with ERC20 | OK |
| Cannot purchase when expired | - not tested as UI doesn't allows it, will be tested when working on a bot |
| Cannot purchase if purchased already | OK |
| Can cancel listing | OK |
| **Bid** | |
| Can create a bid | OK |
| Can bid for a specific price | OK |
| Can bid with a specific expiration date | OK |
| Can bid with an ERC20 | OK |
| Can accept bid with ERC20 | OK |
| Cannot accept bid when expired | - not tested as UI doesn't allows it, will be tested when working on a bot |
| Cannot accept bid if purchased already | OK |
| Can cancel bid | OK |
| **Auction** | |
| Can create an auction | OK |
| Can participate in an auction | OK |
| Can finalize an auction | OK |
| **Asset** | |
| Can mint an asset | OK | 
| Can lazymint an asset | OK | 
| Can upload an image | OK | 
| Can upload an animation | OK | 
| Can upload a private content | OK | 
| Can set royalties | OK | 
| Can add traits | OK | 
| **Account** | | 
| Can update account | OK |
| Can upload image | OK |
| Can upload cover | OK |
| Can remove email/name.. | OK |

### CODE REVIEW

File: packages/react/src/useAcceptAuction.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      invariant(signer, ErrorMessages.SIGNER_FALSY)
+      setActiveProcess(AcceptAuctionStep.INITIAL)
+      try {
+        await client.exchange.acceptAuctionHighestBid(
+          auctionId,
+          signer,
+          (status) => {
+            switch (status.type) {
+              case 'RESOLVE_BEST_BID':
+                return setActiveProcess(AcceptAuctionStep.RESOLVE_BEST_BID)
```
Comment (ismailToyran) [inline packages/react/src/useAcceptAuction.ts:42]: Might need to add new case for this one on modal

File: packages/core/src/uploader/index.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const formData = new FormData()
+    formData.append('file', file)
+    formData.append('protected', isPrivate ? 'true' : 'false')
+    const response = await fetch(this.uploadEndpoint, {
+      method: 'POST',
+      body: formData,
+    })
+    const result = await response.json()
+    invariant(response.ok, `Upload failed: ${result.error}`)
+    return `ipfs://${result.cid}`
```
Comment (ismailToyran) [inline packages/core/src/uploader/index.ts:39]: Just double checking, I've reviewed all commits 1by1
This one was addition, is it intended? (talking about `ipfs://` addition)

**Review by ismailToyran — APPROVED** [review]
This is a big baby 😄 
Reviewed commit by commit.
Everything seems to be good, unless I've missed something.
I've created 2 conversations just to double check, but again rest seems great.
Really liked the react package 😃 Great job 🎉

File: packages/react/src/useAcceptAuction.ts (hunk 1 above)
Comment (antho1404) [reply packages/react/src/useAcceptAuction.ts:42]: Yes indeed, there is a new step, it is not mandatory as this action is fast for now but with the TODO here https://github.com/liteflow-labs/liteflow-js/pull/158/files/d568a3419b147a952cfecdcc4f09de7544db88cc#diff-196345db3b4a08318e68d0452dcc53ff06a024757b1ea06fd767664052c6a358R17 that might take some time

File: packages/core/src/uploader/index.ts (hunk 2 above)
Comment (antho1404) [reply packages/core/src/uploader/index.ts:39]: Yes, we were prefixing with this every time we called the upload function so this is just to simplify and remove a bit of duplication

**Pull request merged**